src/Common/Tests/TestCases/TestEvents.cs

```diff
@@ -8,11 +8,11 @@ public class TestEvents : TestCase
     {
         public override Type MessageType => typeof(ISampleEvent);
 
-        public override void Populate(object instance)
+        public override object CreateInstance(ISerializerFacade serializer)
```

Comment:
This works, but it feels a bit hard to understand how to properly author a test. You're always given a `ISerializerFacade`, but sometimes you use it, and sometimes you don't? It would be nice if the API surface made it a bit clearer somehow.

Reply:
yeah I introduced this as a kind of workaround because not all tests were able to create the necessary instances on their own since they rely on the NServiceBus magic interface implementation feature for that. I'll see if this can be improved somehow.